static/app/utils/useUserTeams.spec.tsx

```diff
@@ -66,4 +73,39 @@ describe('useUserTeams', () => {
     expect(teams.length).toBe(1);
     expect(teams).toEqual(userTeams);
   });
+
+  it('superuser loads all teams', function () {
+    jest.mocked(isActiveSuperuser).mockReturnValue(true);
+
+    const userTeams = [TeamFixture({id: '1', isMember: true})];
+    const nonUserTeams = [TeamFixture({id: '2', isMember: false})];
+    // User teams marked loaded because hasMore is false
+    TeamStore.loadInitialData([...userTeams, ...nonUserTeams], false, null);
+    expect(TeamStore.getState().loadedUserTeams).toBe(true);
+
+    const {result} = reactHooks.renderHook(useUserTeams, {wrapper});
+    const {teams} = result.current;
+
+    expect(teams.length).toBe(2);
+    expect(teams).toEqual(userTeams.concat(nonUserTeams));
+  });
+
+  it('org owner loads all teams', function () {
+    const userTeams = [TeamFixture({id: '1', isMember: true})];
+    const nonUserTeams = [TeamFixture({id: '2', isMember: false})];
+    // User teams marked loaded because hasMore is false
+    TeamStore.loadInitialData([...userTeams, ...nonUserTeams], false, null);
+    expect(TeamStore.getState().loadedUserTeams).toBe(true);
+
+    const organization = OrganizationFixture({
+      access: ['org:admin'],
+    });
+    OrganizationStore.onUpdate(organization, {replace: true});
```

Comment:
Not sure if there's an easier way to update what's in `OrganizationStore` already